### PR TITLE
For #8559 docs(nimbus): Update state and sequence diagrams to use is_dirty flag

### PR DESCRIPTION
**Pull request by eliserichards**
~🚨 do not merge! waiting on #8558~ done

Because

- We want to use the `is_dirty` flag instead of Dirty publish status

This commit

- Updates the state diagrams to remove the Dirty publish status
- Removes Dirty publish status from sequence diagrams
- Adds `is_dirty` flag to sequence diagrams

image
 
state-diagrams-v3

### CODE REVIEW

File: docs/experimenter/README.md [hunk 1]
```diff
@@ -471,7 +470,7 @@ A live rollout can have updates pushed to its state while remaining Live. These
     rect rgb(255,204,255) 
         Note right of Experiment Owner: Owner makes changes in Experimenter
         Experiment Owner->>Experimenter UI: Make updates to live rollout
-        Experimenter UI->>Experimenter Backend: Status: Live <br/> Publish status: Dirty <br/> Status next: <none> <br/> + changelog
+        Experimenter UI->>Experimenter Backend: Status: Live <br/> Publish status: Idle <br/> Status next: <none> <br/> is_dirty: True <br/> + changelog
```
Comment (jaredlockhart) [inline docs/experimenter/README.md:473]: This shows the UI setting the `is_dirty` flag, but why would we change this from the serializer setting it?  We shouldn't need to involve the frontend in it at all?

**Review by jaredlockhart — DISMISSED** [review]
God it's so nice to have these mermaid diagrams now 🎉 🎉 🎉

File: docs/experimenter/README.md (hunk 1 above)
Comment (eliserichards) [reply docs/experimenter/README.md:473]: Good catch. This is one of the things that is changing with the flag vs. publishstatus. I'll update these..

**Review by yashikakhurana — COMMENTED** [review]
Overall it looks good, my suggestion would be to appropriately show the updates we should do for `is_dirty`something like this, thoughts?
image

**Comment by eliserichards** [comment]
Right now it is looking like this: 
image

Do you have a suggestion on making this more clear?

**Comment by yashikakhurana** [comment]
oh perfect, I thought description one is the most recent one

**Review by yashikakhurana — APPROVED** [review]

**Comment by eliserichards** [comment]
Oh sorry, I thought you meant the lifecycle state diagram! I misled you 😅

**Comment by eliserichards** [comment]
Adding "Blocked" until #8558 is reviewed so we can merge them together 🚀

File: docs/experimenter/README.md [hunk 2]
```diff
@@ -471,7 +470,9 @@ A live rollout can have updates pushed to its state while remaining Live. These
     rect rgb(255,204,255) 
         Note right of Experiment Owner: Owner makes changes in Experimenter
         Experiment Owner->>Experimenter UI: Make updates to live rollout
-        Experimenter UI->>Experimenter Backend: Status: Live <br/> Publish status: Dirty <br/> Status next: <none> <br/> + changelog
+        Note over Experimenter Backend: The experiment is updated and is <br/> marked as dirty on the backend
+        Experimenter UI->>Experimenter Backend: Status: Live <br/> Publish status: Idle <br/> Status next: <none>  <br/> + changelog
+        Experimenter Backend->>Experimenter Backend: is_dirty: True
```
Comment (jaredlockhart) [inline docs/experimenter/README.md:475]: Oh yeah that's a great way to represent it 🎉

File: docs/experimenter/README.md [hunk 3]
```diff
@@ -518,12 +519,13 @@ A live rollout can have updates pushed to its state while remaining Live. These
         Note over Experimenter Worker: Worker updates rollout
         Experimenter Worker->>Remote Settings Backend: Check collection (timeout)
         Experimenter Worker->>Experimenter Backend:  Status: Live <br/> Publish status: Idle <br/> Status next: <none> <br/> + changelog
+        Experimenter Backend->>Experimenter Backend: is_dirty: False
```
Comment (jaredlockhart) [inline docs/experimenter/README.md:522]: No, there's no need to explicitly set the value here.  It gets set when the write happens above and then stays there until it is successfully resolved back to False after the update in RS completes.

**Review by jaredlockhart — DISMISSED** [review]
Yes, the little loopty doo is the perfect way to show the backend setting the value.  You just don't need the second one in all the failure cases, take that out and this should be good to go.  Let's land this first and the code second, no need to land them 'together'?

File: docs/experimenter/README.md (hunk 3 above)
Comment (eliserichards) [reply docs/experimenter/README.md:522]: We are showing `is_paused` on these diagrams, which is set in the same places as `is_dirty`. Should we remove `is_paused` from the diagrams as well? (I would prefer to keep both of them)

Comment (eliserichards) [reply docs/experimenter/README.md:522]: oh wait I see what you mean. Misread this ;)

**Pull request merged**